cli/lockfile.rs

```diff
@@ -108,9 +109,29 @@ impl Lockfile {
       None => match maybe_config_file {
         Some(config_file) => {
           if config_file.specifier.scheme() == "file" {
-            let mut path = config_file.specifier.to_file_path().unwrap();
-            path.set_file_name("deno.lock");
-            path
+            if let Some(LockConfig::Bool(lock)) =
+              config_file.clone().to_lock_config()?
+            {
+              if !lock {
+                return Ok(None);
+              }
+            }
+
+            if let Some(LockConfig::PathBuf(lock)) =
+              config_file.clone().to_lock_config()?
+            {
+              config_file
+                .specifier
+                .to_file_path()
+                .unwrap()
+                .parent()
+                .unwrap()
+                .join(lock)
+            } else {
+              let mut path = config_file.specifier.to_file_path().unwrap();
+              path.set_file_name("deno.lock");
+              path
+            }
```

Comment:
We could use a match to reduce cloning and ensure all scenarios are covered (especially when the enum changes):

```suggestion
            match config_file.clone().to_lock_config()? {
              Some(LockConfig::Bool(lock)) => {
                if !lock {
                  return Ok(None);
                }
              }
              Some(LockConfig::PathBuf(lock)) => {
                config_file
                .specifier
                .to_file_path()
                .unwrap()
                .parent()
                .unwrap()
                .join(lock)
              }
              None => {
                let mut path = config_file.specifier.to_file_path().unwrap();
                path.set_file_name("deno.lock");
                path
              }
            }
```

Reply:
That does not work tho.

Reply:
There is a mismatched type.

Reply:
Did you try it?

Reply:
Oh, wait, i see what you mean. One second.

Reply:
I updated the code.

Reply:
Yes, I did. Thanks for helping me :)

Reply:
We could also do a `let path: Option<PathBuf>`, but I don't like it. Do you?

Reply:
@dsherret Done, thanks again!